src/libaktualizr/config/config_test.cc

```diff
@@ -169,19 +169,35 @@ TEST(config, TomlConsistentNonempty) {
   EXPECT_EQ(conf_str1, conf_str2);
 }
 
+static std::vector<std::string> generate_multi_config(TemporaryDirectory &temp_dir) {
+  Utils::writeFile((temp_dir / "a_dir/a.toml"), std::string("[storage]\npath = \"path_a\"\n\n[pacman]\nos = \"os_a\""));
+  Utils::writeFile((temp_dir / "a_dir/z.toml"),
+                   std::string("[storage]\npath = \"path_z\"\n\n[pacman]\nsysroot = \"sysroot_z\""));
+
+  Utils::writeFile((temp_dir / "b_dir/a.toml"), std::string("[storage]\npath = \"cecond_a\""));
+  Utils::writeFile((temp_dir / "b_dir/y.toml"),
+                   std::string("[storage]\npath = \"latest_path\"\n\n[provision]\nprovision_path = \"y_prov_path\""));
+  Utils::writeFile((temp_dir / "b_dir/z.txt"), std::string("[storage]\npath = \"this is path from text file\""));
+
+  return std::vector<std::string>{(temp_dir / "a_dir").c_str(), (temp_dir / "b_dir").c_str()};
```

Comment:
boost::filesystem::path has a `.string()` method which converts into std::string. If you use c_str(), it would probably do two conversions needlessly.